cadence/contracts/connectors/evm/UniswapV3SwapConnectors.cdc

```diff
@@ -501,6 +522,65 @@ access(all) contract UniswapV3SwapConnectors {
                 Burner.burn(<-vault)
             }
         }
+
+        /// OUT amounts: round down to UFix64 precision
+        ///
+        /// Guarantees: convertCadenceAmountToERC20Amount(result) <= original ERC20 amount.
+        access(self) fun _toCadenceOut(
+            _ amt: UInt256,
+            erc20Address: EVM.EVMAddress
+        ): UFix64 {
+            // Bridge utils already floors when going UInt256 -> UFix64
+            return FlowEVMBridgeUtils.convertERC20AmountToCadenceAmount(
+                amt,
+                erc20Address: erc20Address
+            )
+        }
+
+        /// IN amounts: round up to the next UFix64 such that the ERC20 conversion
+        /// (via ufix64ToUInt256) is >= the original UInt256 amount.
+        access(self) fun _toCadenceIn(
+            _ amt: UInt256,
+            erc20Address: EVM.EVMAddress
+        ): UFix64 {
```

Comment:
IMO we should add test cases & ensure cases across extremes are covered since rounding is a common failure point. Will likely require exposing `_toCadenceIn()` these methods as `access(all)`